app/src/gui/components/notebook/index.tsx

```diff
@@ -161,9 +163,22 @@ export default function NotebookComponent({project}: NotebookComponentProps) {
         .then(spec => {
           setUiSpec(spec);
           setViewsets(spec.viewsets);
+          // Set record label based on visible_types
+          if (spec.visible_types && spec.visible_types.length === 1) {
+            const recordType = spec.visible_types[0];
+            console.log('RRRRRRRR-111', recordType);
+            const typeLabel = spec.viewsets[recordType]?.label || recordType;
+            console.log('RRRRRRRR-2222', typeLabel);
+
+            setRecordLabel(typeLabel);
+          } else {
+            setRecordLabel('Record');
+          }
```

Comment:
Looks like you can just derive the recordLabel from the viewsets - which already has a useState - so need for another useState for recordLabel. The component will already re-render when viewset is updated. Can be const in function scope.

Reply:
updated! :-)